src/components/pages/profile/[name]/tabs/MoreTab/Miscellaneous.tsx

```diff
@@ -164,71 +195,199 @@ const Miscellaneous = ({
+    } catch (e) {
+      setError('email', {
+        type: 'submitError',
+        message: t('tabs.more.misc.earnfi.submitError', { ns: 'profile' }),
+      })
+    }
+    setTimeout(() => {
+      clearErrors('email')
+    }, 3000)
+  }
+
   if (!expiryDate) return null
 
   return (
-    <MiscellaneousContainer $isCached={isCachedData || registrationCachedData}>
-      <DatesContainer>
-        {registrationData && (
+    <>
+      <Dialog open={hasEarnifiDialog} variant="blank" onDismiss={() => null}>
+        <Dialog.Heading title={t('tabs.more.misc.earnfi.title', { ns: 'profile' })} />
+        {match(hasCreatedReminder)
+          .with(false, () => (
+            <Form onSubmit={handleSubmit(onSubmit)}>
+              <Typography style={{ textAlign: 'center' }}>
+                <Trans
+                  style={{ textAlign: 'center' }}
+                  i18nKey="tabs.more.misc.earnfi.enterEmail"
+                  ns="profile"
+                  components={{
+                    a: <Outlink href={EARNIFI_OUTLINK} />,
+                  }}
+                />
+              </Typography>
+              <Spacer $height="3" />
+              <Input
+                type="email"
+                id="email"
+                label={t('action.enterEmail')}
+                {...register('email', {
+                  required: t('errors.emailRequired'),
+                  pattern: {
+                    value: emailRegex,
+                    message: t('errors.emailInvalid'),
+                  },
+                })}
+                error={errors.email?.message}
+              />
+              <Spacer $height="3" />
+              <Dialog.Footer
+                leading={
+                  <Button
+                    onClick={() => {
+                      setHasEarnifiDialog(false)
+                      setHasCreatedReminder(false)
+                    }}
+                    colorStyle="accentSecondary"
+                  >
+                    {t('action.cancel')}
+                  </Button>
+                }
+                trailing={
+                  <Button type="submit" disabled={!!errors.email} loading={isSubmitting}>
+                    {t('action.continue')}
+                  </Button>
+                }
+              />
+            </Form>
+          ))
+          .with(true, () => (
+            <>
+              <div style={{ width: '500px', textAlign: 'center' }}>
+                {t('tabs.more.misc.earnfi.emailConfirmation', { ns: 'profile' })}
+              </div>
+              <Dialog.Footer
+                trailing={
+                  <Button
+                    onClick={() => {
+                      setHasEarnifiDialog(false)
+                      setHasCreatedReminder(false)
+                    }}
+                  >
+                    {t('action.close')}
+                  </Button>
+                }
+              />
+            </>
+          ))
+          .exhaustive()}
+      </Dialog>
```

Comment:
probs want to pull this out into a separate component

Reply:
done